### PR TITLE
test: upload test results to Datadog if available

### PR DESCRIPTION
**Pull request by luphaz**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [x] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This PR aims to upload junit test reports generated by Gingko to Datadog

## Code Quality Checklist

- [x] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: api/v1beta1/disruption_types.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ type Reporting struct {
 
 // EmbeddedChaosAPI includes the library so it can be statically exported to chaosli
 //
-//go:embed *
+//go:embed *.go
```
Comment (luphaz) [inline api/v1beta1/disruption_types.go:104]: Not sure but building is not at all possible if I do not change this 🤔

File: .circleci/config.yml [hunk 2]
```diff
@@ -111,8 +111,8 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Install golangci-lint kubebuilder+env helm
-          command: make -j4 install-lint-deps install-kubebuilder install-helm install-controller-gen
+          name: Install binaries
+          command: make -j5 install-lint-deps install-kubebuilder install-helm install-controller-gen install-datadog-ci
```
Comment (taihuynh167) [inline .circleci/config.yml:115]: `j5` is it a type of instance?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -149,6 +149,26 @@ The project contains end-to-end test which are meant to run against a real Kuber
 
 The end-to-end tests will create a set of dummy pods in the `default` namespace of the Lima cluster that will be used to inject different kind of failures and make different assertions against those injections.
 
+### Uploading test results to datadog
+
+In case you have a Datadog account and want to push the tests results to it, you can do the following:
```
Comment (taihuynh167) [inline CONTRIBUTING.md:154]: Do I want to push the test results?
----
When this runs in Gitlab, what key does it use to push results to DD? A team account API key?

**Review by taihuynh167 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (luphaz) [reply CONTRIBUTING.md:154]: Per documentation [here](https://docs.datadoghq.com/continuous_integration/tests/junit_upload/?tab=linux#uploading-test-reports):

> for example, `local` when uploading results from a developer workstation, or ci when uploading them from a CI provider) in the DD_ENV environment variable. For example:

I followed such guidelines.

> 1. Do I want to push the test results?

Up to you TBH, if it helps detecting issue faster and giving timing comparison upon us, etc why you wouldn't want do it?

> 2. When this runs in Gitlab, what key does it use to push results to DD?

This does not run in GitLab but in CircleCI. The key used is a team key indeed.
That I'm also using locally TBH as I'm NOT allowed to create an API key (only an application key)

---

The keys does not really matter for the underlying instrumentation that is git based, it's only needed to have a key with enough permissions to upload an xml file.

Even if not ideal, I think it's OK for now

I still asked the question to have guidance around this.

File: .circleci/config.yml (hunk 2 above)
Comment (luphaz) [reply .circleci/config.yml:115]: `-j` is a make flag that allows to run several make target in [parallel](https://www.gnu.org/software/make/manual/html_node/Parallel.html).

Here we install all program in parallel (low CPU, better to have high concurrency for network calls)

**Pull request merged**